scipy/signal/ltisys.py

```diff
@@ -275,168 +277,144 @@ class lti(object):
+                raise ValueError('Needs 2, 3 or 4 arguments.')
+        # __new__ was called from a subclass, let it call its own functions
+        return super(lti, cls).__new__(cls)
 
-    def __repr__(self):
-        """
-        Canonical representation using state-space to preserve numerical
-        precision and any MIMO information
+    def __init__(self, *system):
+        """Initialize the `lti` baseclass.
+
+        The heavy lifting is done by the subclasses.
         """
-        return '{0}(\n{1},\n{2},\n{3},\n{4}\n)'.format(
-            self.__class__.__name__,
-            repr(self.A),
-            repr(self.B),
-            repr(self.C),
-            repr(self.D),
-            )
+        self.inputs = None
+        self.outputs = None
 
     @property
     def num(self):
-        return self._num
+        return self.to_tf().num
 
     @num.setter
-    def num(self, value):
-        self._num = value
-        self._update(2)
+    def num(self, num):
+        obj = self.to_tf()
+        obj.num = num
+        source_class = type(self)
+        self._copy(source_class(obj))
 
     @property
     def den(self):
-        return self._den
+        return self.to_tf().den
 
     @den.setter
-    def den(self, value):
-        self._den = value
-        self._update(2)
+    def den(self, den):
+        obj = self.to_tf()
+        obj.den = den
+        source_class = type(self)
+        self._copy(source_class(obj))
 
     @property
     def zeros(self):
-        return self._zeros
+        return self.to_zpk().zeros
 
     @zeros.setter
-    def zeros(self, value):
-        self._zeros = value
-        self._update(3)
+    def zeros(self, zeros):
+        obj = self.to_zpk()
+        obj.zeros = zeros
+        source_class = type(self)
+        self._copy(source_class(obj))
 
     @property
     def poles(self):
-        return self._poles
+        return self.to_zpk().poles
 
     @poles.setter
-    def poles(self, value):
-        self._poles = value
-        self._update(3)
+    def poles(self, poles):
+        obj = self.to_zpk()
+        obj.poles = poles
+        source_class = type(self)
+        self._copy(source_class(obj))
 
     @property
     def gain(self):
-        return self._gain
+        return self.to_zpk().gain
 
     @gain.setter
-    def gain(self, value):
-        self._gain = value
-        self._update(3)
+    def gain(self, gain):
+        obj = self.to_zpk()
+        obj.gain = gain
+        source_class = type(self)
+        self._copy(source_class(obj))
 
     @property
     def A(self):
-        return self._A
+        return self.to_ss().A
 
     @A.setter
-    def A(self, value):
-        self._A = value
-        self._update(4)
+    def A(self, A):
+        obj = self.to_ss()
+        obj.A = A
+        source_class = type(self)
+        self._copy(source_class(obj))
 
     @property
     def B(self):
-        return self._B
+        return self.to_ss().B
```

Comment:
Yes. The previous lti class did the same thing, though in a less explicit way. This function is overwritten in the StateSpace subclass, so that this is only a concern if you access/alter elements that are not part of your current system (like the B matrix of a TransferFunction)